dlfcn.go

```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: 2022 The Ebitengine Authors
 
-//go:build darwin || freebsd || linux
+//go:build darwin || freebsd || (linux && !faketime)
```

Comment:
```go
//go:build (darwin || freebsd || linux) && !faketime
```

seems better. We don't have to depend on the fact that GOOS is linux on the playground.